src/ucp/core/ucp_ep.c

```diff
@@ -1048,15 +1048,32 @@ ucp_ep_config_lane_is_dst_rsc_index_equal(const ucp_ep_config_key_t *key1,
            (key1->lanes[lane1].dst_rsc_index == key2->lanes[lane2].dst_rsc_index);
 }
 
+static int
+ucp_ep_config_lane_is_local_info_equal(const ucp_ep_config_key_t *key1,
+                                       ucp_lane_index_t lane1,
+                                       const ucp_ep_config_key_t *key2,
+                                       ucp_lane_index_t lane2)
+{
+    return (key1->lanes[lane1].rsc_index == key2->lanes[lane2].rsc_index) &&
+           (key1->lanes[lane1].path_index == key2->lanes[lane2].path_index);
+}
+
 int ucp_ep_config_lane_is_peer_equal(const ucp_ep_config_key_t *key1,
                                      ucp_lane_index_t lane1,
                                      const ucp_ep_config_key_t *key2,
                                      ucp_lane_index_t lane2)
 {
-    return (key1->lanes[lane1].rsc_index  == key2->lanes[lane2].rsc_index) &&
-           ucp_ep_config_lane_is_dst_rsc_index_equal(key1, lane1, key2, lane2) &&
-           (key1->lanes[lane1].path_index == key2->lanes[lane2].path_index) &&
-           (key1->lanes[lane1].dst_md_index  == key2->lanes[lane2].dst_md_index);
+    if ((key1->flags & UCP_EP_CONFIG_FLAG_FAKE) ||
+        (key2->flags & UCP_EP_CONFIG_FLAG_FAKE)) {
+        /* if at least one configuration is a fake one, check only local
+         * characteristics */
+        return ucp_ep_config_lane_is_local_info_equal(key1, lane1, key2, lane2);
```

Comment:
can it lead to the wrong assumption on configs intersections if we take into account only local info?

Reply:
we consider FAKE config as only locally valid - will add a comment about it
so, when only the local part is the same as the local part of another FAKE config or normal config, we consider such configs are intersected

Reply:
why? if destination md/rsc are different we can not assume they are the same